stellarphot/settings/tests/test_custom_widgets.py

```diff
@@ -816,44 +839,420 @@ def test_title_format(self):
+
+        # Create the review widget and check that the error is raised
+        with pytest.raises(
+            ValueError, match="The camera setting saved in the working directory is not"
+        ):
+            ReviewSettings([Camera])
+
+    def test_error_when_setting_has_no_saved_or_default_setting(self):
+        # Make a review widget with a setting that has no saved or default setting
+        # like Camera and check that an error is raised.
+
+        review_settings = ReviewSettings([Camera])
+
+        assert SaveStatus.SETTING_NOT_SAVED in review_settings._container.titles[0]
+
+    def test_setting_selected_item_to_none_does_not_save(self):
+        # The intent of this is to test an edge case where the user selects an item
+        # from the dropdown and then selects "None" from the dropdown. The selected
+        # item should be saved, but there should not be another save when None is
+        # selected.
+        saved = SavedSettings()
+        camera = Camera(**TEST_CAMERA_VALUES)
+        saved.add_item(camera)
+        wd_settings = PhotometryWorkingDirSettings()
+        wd_set_path = Path(wd_settings.partial_settings_file)
+        review_settings = ReviewSettings([Camera])
+
+        # Get the modification time of the working directory settings file
+        mtime = wd_set_path.stat().st_mtime
+
+        assert (
+            review_settings._setting_widgets[0]._widget._choose_existing.value == camera
+        )
+        review_settings._setting_widgets[0]._widget._choose_existing.value = "none"
+
+        # Check that the working directory settings file has not been modified
+        assert wd_set_path.stat().st_mtime == mtime
+
+    def test_clicking_tab_with_already_saved_settings_updates_badge(self):
+        # Check that if there are already settings saved to the working directory
+        # and the user clicks on the tab, the badge is updated to reflect that the
+        # setting has been saved.
+
+        # To set this up we need to save settings to the working directory AND to the
+        # saved user settings.
+        wd_settings = PhotometryWorkingDirSettings()
+        camera = Camera(**TEST_CAMERA_VALUES)
+        observatory = Observatory(**DEFAULT_OBSERVATORY_SETTINGS)
+        passbands = PassbandMap(**DEFAULT_PASSBAND_MAP)
+        wd_settings.save(
+            PartialPhotometrySettings(
+                camera=camera, observatory=observatory, passband_map=passbands
+            )
+        )
+        saved = SavedSettings()
+        saved.add_item(camera)
+        saved.add_item(observatory)
+        saved.add_item(passbands)
+
+        # Make the review widget
+        review_settings = ReviewSettings([Camera, Observatory, PassbandMap])
+        for title in review_settings._container.titles:
+            assert SaveStatus.SETTING_SHOULD_BE_REVIEWED in title
+
+        # Click on each tab, starting with the last one, to make sure a change in
```

Comment:
I suppose selecting the index is the programmatic equivalent of clicking, but seems odd wording (no need to change, just took me a bit to process this).